pkg/controller/statusmanager/status_manager.go

```diff
@@ -36,6 +37,8 @@ type StatusManager struct {
 	name    string
 	version string
 
+	statusUpdateLock sync.Mutex
```

Comment:
The problem isn't just with status updates. Eg, it's possible that one thread is calling `SetFromPods` while another thread is calling `SetDaemonSets`, but we're not synchronizing the access to `status.daemonSets`. We need to mutex the whole `StatusManager` and all of its methods.

Reply:
> The problem isn't just with status updates. Eg, it's possible that one thread is calling `SetFromPods` while another thread is calling `SetDaemonSets`, but we're not synchronizing the access to `status.daemonSets`. We need to mutex the whole `StatusManager` and all of its methods.

Correct me if I'm wrong, but `SetDaemonSets`, `SetDeployments` and `SetRelatedObjects` are only called from here. Which means that we can rely on them and `SetFromPods` always being called sequentially from only one thread. 

I guess the concern could be that multiple threads step on each others toes by calling `SetFromPods` individually (and one of them not having the most recent state and thus getting it wrong). But, seeing as how `set` now is synchronized it shouldn't lead to any fatal "errors" (i.e one calling `Create` instead of `Update`)? Whatever status is set at any given moment in time by any of these X threads (and might be incorrect) should be re-updated any moment later once the new thread acquires the lock and performs his update with the most up-to-date state? 

The only thing I can come to think of, as I am writing this, is `SetDegraded` and `SetNotDegraded` modifying `status.failing` unsynchronized. I will thus update my PR with a fix for this.

Reply:
> But, seeing as how `set` now is synchronized it shouldn't lead to any fatal "errors" (i.e one calling `Create` instead of `Update`)?

Note that golang does not give you any more automatic thread-safety than C does. If the `operconfig_controller` thread calls `SetDaemonSets` while the `pod_controller` thread calls `SetFromPods`, without using any synchronization primitives, then the `SetFromPods` thread could see literally any value in `status.daemonSets` (including in particular an invalid pointer).

Reply:
Indeed, I overlooked that. I have a (maybe not so pretty) solution to that which I will push and let you review.